x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/IndexInputStats.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.searchablesnapshots.cache;
+
+import org.elasticsearch.xpack.searchablesnapshots.cache.CacheDirectory.CacheBufferedIndexInput;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.LongAdder;
+import java.util.function.LongConsumer;
+
+/**
+ * {@link IndexInputStats} records stats for a given {@link CacheBufferedIndexInput}.
+ */
+public class IndexInputStats {
+
+    static final double SEEKING_THRESHOLD = 0.25d;
```

Comment:
One of the big open questions about seeking is whether we are seeking forwards in small enough steps that we should have just requested a larger contiguous chunk of the blob up-front. Today the implementation is based on fixed-size chunks, so I think it makes sense to set a fixed-size threshold for what we consider to be a "large" seek. I would guess something like 8MB or 16MB would be a good start.

We could of course move to a chunk size that is proportional to the size of the underlying file, and in that case it makes more sense for the threshold to be proportional to the file size too.

Reply:
This is a good observation. I pushed 51c4f43a5753d38d81beabb83024c9268b99913d to change the seeking threshold to 8Mb, let me know what you think.

Reply:
:+1: